### ISSUE TITLE
Aboutstop-gradient operation

### ISSUE THREAD

**Issue by ghost** [kunheek/style-aware-discriminator#2]
Hi, thank you for sharing your code. Can you please tell me how to stop the gradient  when training the generator in your code?  Looking forward to your reply, thanks!

**Comment by kunheek** [comment]
Hi. Thank you for your interest in our work!

I'm not sure if you are talking about the generator or style code. To implement the stop gradient described in paper, I used `torch.no_grad()` or `tensor.detach()`
Here's the actual implementations:
- [Sampling style code from the image (Fig. 1(a) top](https://github.com/kunheek/style-aware-discriminator/blob/d3fb0db54288a0afc1f48d9ac19f7903264816c1/model/networks/prototypes.py#L71).
- [Sampling style code from the image (Fig. 1(a) bottom](https://github.com/kunheek/style-aware-discriminator/blob/d3fb0db54288a0afc1f48d9ac19f7903264816c1/model/model.py#L143)
- [Fig. 1(c)](https://github.com/kunheek/style-aware-discriminator/blob/d3fb0db54288a0afc1f48d9ac19f7903264816c1/model/model.py#L230)

If you are talking about freezing other networks during the generator update, I set `requires_grad_(False)`. You can find it [here](https://github.com/kunheek/style-aware-discriminator/blob/d3fb0db54288a0afc1f48d9ac19f7903264816c1/model/model.py#L233).

I hope this helps you.

**Comment by ghost** [comment]
Got it. I'm talking about the style code. Thanks a lot!

**Comment by ghost** [comment]
Sorry to bother again.
This model keeps the background of images very well, and I think this may be because the style codes are updated separately. If the style code is updated jointly, which code needs to be changed in addition to the three mentioned above?

**Comment by kunheek** [comment]
Actually, the style code cannot be trained jointly. It's technically possible, but it will generate unrealistic samples.

**Comment by ghost** [comment]
Got it. Thanks for your reply!
